### DISCUSSION TITLE
Just started setting up Storybook in a fresh project and argTypesRegex is having no effect. What am I missing?

### DISCUSSION THREAD

**Discussion by iansan5653** [storybookjs/storybook#15911] [Help] [answered]
I'm trying to set up Storybook in a new TypeScript React project. I used `create-react-app` and `sb init` to set everything up, so pretty much standard there. I have a super simple `Button` component:

```tsx
import { ReactElement } from "react";

export interface ButtonProps {
  text: string;
  onClick: () => void;
}

export default function Button(props: ButtonProps): ReactElement {
  return <button onClick={() => props.onClick()}>{props.text}</button>;
}
```

I set up a basic story with args:

```tsx
import { Meta, Story } from "@storybook/react";
import Button, { ButtonProps } from "components/Button";

const meta: Meta = {
  title: "Button"
};

export default meta;

const Template: Story<ButtonProps> = args => <Button {...args} />;

export const Default = Template.bind({});
Default.args = {
  text: "Button Text"
};
```

The story loads, renders, and I can customize the button text. Great!

Now for Actions, I expected it to 'just work' because my `preview.js` file has a matcher that should match `onClick`.
```js
export const parameters = {
  actions: { argTypesRegex: "^on[A-Z].*" },
};
```

However, something is missing here and I'm not sure what it is. When I click the button, instead of logging an action it just fails with `props.onClick is not a function`.

In case it matters, my `main.js` is unmodified from the default:
```js
module.exports = {
  "stories": [
    "../src/**/*.stories.tsx"
  ],
  "addons": [
    "@storybook/addon-links",
    "@storybook/addon-essentials",
    "@storybook/preset-create-react-app"
  ]
}
```

Any advice is greatly appreciated.

**Comment by iansan5653** [comment, marked as answer]
Solved this pretty quick actually 😅. Turns out you need to define the `component` property on the metadata for this to work. This is somewhat unexpected though since the Controls addon doesn't need that at all. The [documentation code sample](https://storybook.js.org/docs/react/essentials/actions#automatically-matching-args) has this defined, but it's not called out anywhere in the documentation text - maybe it should be noted if this is expected behavior?

**Reply by shilman** [reply]
cc @jonniebigodes
